examples/kubernetes/connectivity-check/resources.cue

```diff
@@ -224,8 +224,17 @@ egressCNP: [ID=_]: _cnp & {
 					matchLabels: {
 						"k8s:io.kubernetes.pod.namespace": "kube-system"
 						"k8s:k8s-app":                     "kube-dns"
+					},
+					// Allows the tests to pass when NodeLocal DNSCache exists and is not configured with hostNetwork=true.
+					matchLabels: {
+						"k8s:io.kubernetes.pod.namespace": "kube-system"
+						"k8s:k8s-app":                     "node-local-dns"
 					}
```

Comment:
This is not not correct in term of syntax, please refer to the diff below. Kindly note that this is just to demonstration the changes based on your patch.

Locally, you can run `make -C examples/kubernetes/connectivity-check all` to generate all manifest again.

<details>

```dfff
diff --git a/examples/kubernetes/connectivity-check/connectivity-check-hostport.yaml b/examples/kubernetes/connectivity-check/connectivity-check-hostport.yaml
index 8f6c27391a..6a1221a455 100644
--- a/examples/kubernetes/connectivity-check/connectivity-check-hostport.yaml
+++ b/examples/kubernetes/connectivity-check/connectivity-check-hostport.yaml
@@ -1730,6 +1730,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -1770,6 +1773,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -1808,6 +1814,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -1855,6 +1864,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -1899,6 +1911,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -1943,6 +1958,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -1987,6 +2005,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
diff --git a/examples/kubernetes/connectivity-check/connectivity-check-internal.yaml b/examples/kubernetes/connectivity-check/connectivity-check-internal.yaml
index 0db2720a6e..f8979791bc 100644
--- a/examples/kubernetes/connectivity-check/connectivity-check-internal.yaml
+++ b/examples/kubernetes/connectivity-check/connectivity-check-internal.yaml
@@ -833,6 +833,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -873,6 +876,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
diff --git a/examples/kubernetes/connectivity-check/connectivity-check-proxy.yaml b/examples/kubernetes/connectivity-check/connectivity-check-proxy.yaml
index fa12ba711d..b5be2121bc 100644
--- a/examples/kubernetes/connectivity-check/connectivity-check-proxy.yaml
+++ b/examples/kubernetes/connectivity-check/connectivity-check-proxy.yaml
@@ -686,6 +686,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -730,6 +733,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -774,6 +780,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -818,6 +827,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
diff --git a/examples/kubernetes/connectivity-check/connectivity-check-quarantine.yaml b/examples/kubernetes/connectivity-check/connectivity-check-quarantine.yaml
index 488ddd57f6..fb637a1f18 100644
--- a/examples/kubernetes/connectivity-check/connectivity-check-quarantine.yaml
+++ b/examples/kubernetes/connectivity-check/connectivity-check-quarantine.yaml
@@ -504,6 +504,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -545,6 +548,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -586,6 +592,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -627,6 +636,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
diff --git a/examples/kubernetes/connectivity-check/connectivity-check-single-node.yaml b/examples/kubernetes/connectivity-check/connectivity-check-single-node.yaml
index 693d8366b3..5649871284 100644
--- a/examples/kubernetes/connectivity-check/connectivity-check-single-node.yaml
+++ b/examples/kubernetes/connectivity-check/connectivity-check-single-node.yaml
@@ -613,6 +613,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -653,6 +656,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -691,6 +697,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
diff --git a/examples/kubernetes/connectivity-check/connectivity-check.yaml b/examples/kubernetes/connectivity-check/connectivity-check.yaml
index f2be26ae42..1429f0bad9 100644
--- a/examples/kubernetes/connectivity-check/connectivity-check.yaml
+++ b/examples/kubernetes/connectivity-check/connectivity-check.yaml
@@ -945,6 +945,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -985,6 +988,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
@@ -1023,6 +1029,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
diff --git a/examples/kubernetes/connectivity-check/connectivity-debug-tools.yaml b/examples/kubernetes/connectivity-check/connectivity-debug-tools.yaml
index 855ef6af91..7b60b401e2 100644
--- a/examples/kubernetes/connectivity-check/connectivity-debug-tools.yaml
+++ b/examples/kubernetes/connectivity-check/connectivity-debug-tools.yaml
@@ -68,6 +68,9 @@ spec:
     - matchLabels:
         k8s:io.kubernetes.pod.namespace: kube-system
         k8s:k8s-app: kube-dns
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: node-local-dns
   - toPorts:
     - ports:
       - port: "5353"
diff --git a/examples/kubernetes/connectivity-check/echo-servers.cue b/examples/kubernetes/connectivity-check/echo-servers.cue
index 3c506bbace..7671c21931 100644
--- a/examples/kubernetes/connectivity-check/echo-servers.cue
+++ b/examples/kubernetes/connectivity-check/echo-servers.cue
@@ -40,8 +40,9 @@ deployment: "echo-b-host": _echoDeploymentWithHostPort & {
 }
 
 ingressL7Policy: {
-	_allowDNS: true
-	_port:     *"8080" | string
+	_allowDNS:     true
+	_localDNSCIDR: ""
+	_port:         *"8080" | string
 	_rules: [{
 		toPorts: [{
 			ports: [{
diff --git a/examples/kubernetes/connectivity-check/resources.cue b/examples/kubernetes/connectivity-check/resources.cue
index 4676a79300..4fa34cba55 100644
--- a/examples/kubernetes/connectivity-check/resources.cue
+++ b/examples/kubernetes/connectivity-check/resources.cue
@@ -206,7 +206,8 @@ _cnp: {
 egressCNP: [ID=_]: _cnp & {
 	_name: ID
 	_rules: [...{}]
-	_allowDNS: *true | false
+	_allowDNS:     *true | false
+	_localDNSCIDR: "169.254.25.10/32" | string
 
 	// Implicitly open DNS visibility if FQDN rule is specified.
 	_enableDNSVisibility: *false | true
@@ -220,21 +221,24 @@ egressCNP: [ID=_]: _cnp & {
 	if _allowDNS {
 		spec: egress: _rules + [
 				{
-				toEndpoints: [{
-					matchLabels: {
-						"k8s:io.kubernetes.pod.namespace": "kube-system"
-						"k8s:k8s-app":                     "kube-dns"
-					},
+				toEndpoints: [
+					{
+						matchLabels: {
+							"k8s:io.kubernetes.pod.namespace": "kube-system"
+							"k8s:k8s-app":                     "kube-dns"
+						}},
 					// Allows the tests to pass when NodeLocal DNSCache exists and is not configured with hostNetwork=true.
-					matchLabels: {
-						"k8s:io.kubernetes.pod.namespace": "kube-system"
-						"k8s:k8s-app":                     "node-local-dns"
-					}
-				}]
-				// Allows the tests to pass when NodeLocal DNSCache is configured with hostNetwork=true.
-				toCIDR: [
-					169.254.25.10/32
+					{
+						matchLabels: {
+							"k8s:io.kubernetes.pod.namespace": "kube-system"
+							"k8s:k8s-app":                     "node-local-dns"
+						}},
 				]
+				// Allows the tests to pass when NodeLocal DNSCache is configured with hostNetwork=true.
+				if _localDNSCIDR != "" {
+					toCIDR: [_localDNSCIDR]
+				}
+
 				toPorts: [{
 					ports: [{
 						port:     "53"
```

</details>